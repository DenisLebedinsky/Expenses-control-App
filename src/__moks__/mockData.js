export default [
  {
    id: 1, name: 'January', expenses: 4000, incomes: 2400, amt: 2400,
  },
  {
    id: 2, name: 'February', expenses: 2222, incomes: 1398, amt: 2210,
  },
  {
    id: 3, name: 'March', expenses: 2000, incomes: 9800, amt: 2290,
  },
  {
    id: 4, name: 'April', expenses: 2780, incomes: 3908, amt: 2000,
  },
  {
    id: 5, name: 'June', expenses: 1890, incomes: 4800, amt: 2181,
  },
  {
    id: 6, name: 'July', expenses: 2390, incomes: 3800, amt: 2500,
  },
  {
    id: 7, name: 'August', expenses: 3490, incomes: 4300, amt: 2100,
  },
];