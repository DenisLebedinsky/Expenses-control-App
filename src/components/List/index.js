import React from 'react'
import { DebounceInput } from "react-debounce-input";

export default ({ items, edit, del }) => {

  const editItem = (id, type, value) => {
	
    if (typeof edit === 'function') {
      edit(id, type, value)
    }
  }

  const deleteItem = (id) => {
    if (typeof del === 'function') {
      del(id)
    }
  }

  return (
    <div className="container">
      <div className="row">
        <form className="col-md-12" data-testid="list_of_items">
          {items &&
            items.map(item => (
              <div
                className=" d-flex align-items-end justify-content-center"
                key={item.id}
                data-testid={item.id}
              >
                <div className="form-group mr-1">
                  <DebounceInput
                    type="string"
                    className="form-control"
                    value={item.name}
                    onChange={e => editItem(item.id, "name", e.target.value)}
                    debounceTimeout={3000}
                  />
                </div>
                <div className="form-group mr-1">
                  <DebounceInput
                    type="number"
                    className="form-control"
                    value={item.expenses}
                    onChange={e =>
                      editItem(item.id, "expenses", e.target.value)
                    }
                    debounceTimeout={3000}
                  />
                </div>
                <div className="form-group mr-1">
                  <DebounceInput
                    type="number"
                    className="form-control"
                    value={item.incomes}
                    onChange={e =>
                      editItem(item.id, "incomes", e.target.value)
                    }
                    debounceTimeout={3000}
                  />
                </div>

                <div className="form-group">
                  <button
                    className="btn btn-outline-danger"
                    onClick={() => deleteItem(item.id)}
                  >
                    X
                  </button>
                </div>
              </div>
            ))}
        </form>
      </div>
    </div>
  );
}