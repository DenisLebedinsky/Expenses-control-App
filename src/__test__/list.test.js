import React from "react";
import List from "../components/List";
import {
  render,
  cleanup,
  waitForElement,
  fireEvent
} from "@testing-library/react";

import mockData from "../__moks__/mockData";

describe("List", () => {
  afterEach(cleanup);

  it("test for render data", async () => {
    const { container, getByTestId } = render(<List items={mockData} />);

    await waitForElement(() => getByTestId("list_of_items"));

    await waitForElement(() => getByTestId("3"));
    expect(container.firstChild).toMatchSnapshot();
  });




  const setup = () => {

    const utils = render(<List items={mockData} />);
    const input = utils.getByTestId("2").firstChild.firstChild;

    return {
      input,
      ...utils
    };
  };

  afterEach(cleanup);

  it("test input method", () => {
    const { input, handleChange } = setup();

    expect(input.value).toBe("February");
    fireEvent.change(input, { target: { value: "test input_name" } });
		
		expect(handleChange).toHaveBeenCalled();
    expect(input.value).toBe("test input_name");
  });
});
