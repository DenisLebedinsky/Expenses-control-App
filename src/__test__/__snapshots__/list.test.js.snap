// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List test for render data 1`] = `
<div
  class="container"
>
  <div
    class="row"
  >
    <form
      class="col-md-12"
      data-testid="list_of_items"
    >
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="1"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="January"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="4000"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="2400"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="2"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="February"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="2222"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="1398"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="3"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="March"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="2000"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="9800"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="4"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="April"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="2780"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="3908"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="5"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="June"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="1890"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="4800"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="6"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="July"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="2390"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="3800"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
      <div
        class=" d-flex align-items-end justify-content-center"
        data-testid="7"
      >
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="string"
            value="August"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="3490"
          />
        </div>
        <div
          class="form-group mr-1"
        >
          <input
            class="form-control"
            disabled=""
            type="number"
            value="4300"
          />
        </div>
        <div
          class="form-group"
        >
          <button
            class="btn btn-outline-danger"
          >
            X
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
